internal/operands/common-templates/reconcile.go

```diff
@@ -123,7 +132,59 @@ func reconcileEditRole(request *common.Request) (common.ResourceStatus, error) {
 		})
 }
 
-func reconcileTemplatesFuncs(request *common.Request) []common.ReconcileFunc {
+func reconcileOlderTemplates(request *common.Request) ([]common.ReconcileFunc, error) {
+	// Append functions to take ownership of previously deployed templates during an upgrade
+	var existingTemplates templatev1.TemplateList
+	templatesSelector := func() labels.Selector {
```

Comment:
You could move this into its own function instead of an inline definition.
Like a `buildTemplateSelector`, then you can also handle the errors without panics (but they seem like assertions so it should be fine).
Tbh. I don't understand right now how this actually works, as LabelSelector takes an interface but you pass it a `func() labels.Selector`.

Reply:
I don't pass a function, I define it an immediately execute it, notice the `()` in line 161. I do it just to scope the creation and definition of the label selector (makes it more readable IMO).

To you other point, I don't think creating a function is required, because the label selectors here are specific enough

Reply:
Ah, I overlooked the ().
In general having functionality inlined like this and immediately executed is something I rarely see in Golang so far, I think that's more a JavaScript thing.
To me it's more confusing because I didn't expect it, but if it's a common thing in the project I don't disagree.

I'm not sure right now if there is a runtime difference between the two (how the compiler optimizes inline functions), but it shouldn't matter. Mainly I was surprised because I'm only used to inline functions like this for defers or to pass the function.

Reply:
I also find it a little unusual. No need to change this now, but in the future you can create a new scope using `{ }`.
```
var templatesSelector labels.Selector
{
    tplSelector := labels.NewSelector()
    ...
    templatesSelector = tplSelector
}
```

Reply:
You are right! I forgot that is even an option